### PR TITLE
Test skeletons for transport module - Subtask #792 - Closes #838

### LINKED ISSUE #838: Test skeletons for transport module
To make writing unit tests for missing coverage easier, we should write skeletons first. This issue tracks the skeleton implementation for transport module.

Related to #792

### PR DESCRIPTION
**Pull request by Tschakki**
related to #792
Closes #838

### CODE REVIEW

File: test/unit/modules/transport.js [hunk 1]
```diff
@@ -0,0 +1,892 @@
...
+
+					describe('when query.updateType = 1', function () {
+
+						it('should call modules.peers.remove');
+
+						it('should call modules.peers.remove with query.peer');
+					});
+				});
+				
+				describe('when upDateResult = false', function () {
```
Comment (Isabello) [inline test/unit/modules/transport.js]: casing. - `updateResult`

File: test/unit/modules/transport.js [hunk 2]
```diff
@@ -0,0 +1,892 @@
...
+					
+					it('should call callback with error = new PeerUpdateError(updateResult, failureCodes.errorMessages[updateResult])');
+				});
+				
+				describe('when upDateResult = true', function () {
```
Comment (Isabello) [inline test/unit/modules/transport.js]: as previous

File: test/unit/modules/transport.js [hunk 3]
```diff
@@ -0,0 +1,892 @@
...
+			});
+			
+			it('should call modules.transactions.processUnconfirmedTransaction');
+			
+			it('should call modules.transactions.processUnconfirmedTransaction with transaction');
+			
+			it('should call modules.transactions.processUnconfirmedTransaction with true');
+			
+			describe('when modules.transactions.processUnconfirmedTransaction fails', function () {
+				it('should call library.logger.debug');
```
Comment (Isabello) [inline test/unit/modules/transport.js]: new line

File: test/unit/modules/transport.js [hunk 4]
```diff
@@ -0,0 +1,892 @@
...
+				describe('when it does not throw', function () {
+					
+					describe('when query.block is defined', function () {
+						
+						it('should call bson.deserialize');
+						
+						it('should call bson.deserialize with Buffer.from(query.block)');
+						
+						describe('block', function () {
+							it('should call modules.blocks.verify.addBlockProperties');
```
Comment (Isabello) [inline test/unit/modules/transport.js]: new line

File: test/unit/modules/transport.js [hunk 5]
```diff
@@ -0,0 +1,894 @@
...
+			
+			it('should call library.schema.validate');
+					
+			it('should call library.schema.validate with query');
+
+			it('should call library.schema.validate with schema.signatures');
+
+			describe('when library.schema.validate fails', function () {
+				
+				it('should call series callback with "Invalid signatures body"');
```
Comment (MaciejBaj) [inline test/unit/modules/transport.js]: with error = "...

File: test/unit/modules/transport.js [hunk 6]
```diff
@@ -0,0 +1,894 @@
...
+			
+			describe('when library.schema.validate succeeds', function () {
+				
+				it('should call modules.multisignatures.processSignature');
+				
+				it('should call modules.multisignatures.processSignature with query');
+				
+				describe('when modules.multisignatures.processSignature fails', function (){
+					
+					it('should call callback with "Error processing signature:" + err');
```
Comment (MaciejBaj) [inline test/unit/modules/transport.js]: just error I think

File: test/unit/modules/transport.js [hunk 7]
```diff
@@ -0,0 +1,894 @@
...
+			
+			it('should call modules.transactions.processUnconfirmedTransaction with transaction');
+			
+			it('should call modules.transactions.processUnconfirmedTransaction with true');
+			
+			describe('when modules.transactions.processUnconfirmedTransaction fails', function () {
+				
+				it('should call library.logger.debug');
+				
+				it('should call library.logger.debug with ["Transaction", id].join(" ")');
```
Comment (MaciejBaj) [inline test/unit/modules/transport.js]: with "Transaction ${transaction.id}"

File: test/unit/modules/transport.js [hunk 8]
```diff
@@ -0,0 +1,894 @@
...
+				it('should call callback with error = null');
+				
+				it('should call callback with result = transaction.id');
+			});
+		});
+	});
+	
+	describe('Transport', function () {
+		
+		describe('prototype.headers', function () {
```
Comment (MaciejBaj) [inline test/unit/modules/transport.js]: remove prototype keyword here and in following examples

File: test/unit/modules/transport.js [hunk 9]
```diff
@@ -0,0 +1,894 @@
...
+			describe('when __private.checkInternalAccess fails', function () {
+				
+				it('should call callback wit error = err');
+			});
+			
+			describe('when __private.checkInternalAccess succeeds', function () {
+				
+				describe('updateResult', function () {
+					
+					describe('when query.updateType = 0', function () {
```
Comment (MaciejBaj) [inline test/unit/modules/transport.js]: = 0 (insert)
= 1 (remove)

**Pull request merged**